### PR TITLE
Drop feature flag for v0 cid support

### PR DESCRIPTION
**Pull request by cocreature**
I’ve kept it on the same test infrastructure that we used before that
can test both ways since it seems nicer to keep this consistent with v1.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto [hunk 1]
```diff
@@ -76,14 +76,8 @@ enum CommandDeduplicationType {
 
 // See `daml-lf/spec/contract-id.rst` for more information on contract ID formats.
 message ExperimentalContractIds {
-  ContractIdV0Support v0 = 1;
```
Comment (SamirTalwar-DA) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto:79]: I suggest adding `reserved 1`.

Alternatively, you can renumber `v1` as we haven't released this yet.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractIdIT.scala [hunk 2]
```diff
@@ -30,18 +30,13 @@ import scala.util.{Failure, Success, Try}
...
-    ),
     TestConfiguration(
       description = "v0",
       example = v0Cid,
       accepted = false,
-      isSupported = features => features.contractIds.v0.isNotSupported,
+      // Support for v0 contract ids existed only in sandbox-classic in
+      // SDK 1.18 and older and has been dropped completely.
+      isSupported = _ => true,
       disabledReason = "V0 contract IDs are supported",
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractIdIT.scala:40]: You can just delete these two lines; they have default values.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto (hunk 1 above)
Comment (cocreature) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto:79]: renumbered, doesn’t seem useful to keep it around.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractIdIT.scala (hunk 2 above)
Comment (cocreature) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ContractIdIT.scala]: dropped thanks

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**